src/core_plugins/kibana/public/management/sections/objects/lib/resolve_saved_objects.js

```diff
@@ -0,0 +1,195 @@
+import { SavedObjectNotFound } from 'ui/errors';
+
+async function getSavedObject(doc, services) {
+  const service = services.find(service => service.type === doc._type);
+  if (!service) {
+    return;
+  }
+
+  const obj = await service.get();
+  obj.id = doc._id;
+  return obj;
+}
+
+async function importIndexPattern(doc, indexPatterns, overwriteAll) {
+  // TODO: consolidate this is the code in create_index_pattern_wizard.js
+  const emptyPattern = await indexPatterns.get();
+  Object.assign(emptyPattern, {
+    id: doc._id,
+    title: doc._source.title,
+    timeFieldName: doc._source.timeFieldName,
+  });
+  const newId = await emptyPattern.create(true, !overwriteAll);
+  indexPatterns.cache.clear(newId);
+  return newId;
+}
+
+async function importDocument(obj, doc, overwriteAll) {
+  await obj.applyESResp(doc);
+  return await obj.save({ confirmOverwrite: !overwriteAll });
+}
+
+function groupByType(docs) {
+  const defaultDocTypes = {
+    searches: [],
+    indexPatterns: [],
+    other: [],
+  };
+
+  return docs.reduce((types, doc) => {
+    switch (doc._type) {
+      case 'search':
+        types.searches.push(doc);
+        break;
+      case 'index-pattern':
+        types.indexPatterns.push(doc);
+        break;
+      default:
+        types.other.push(doc);
+    }
+    return types;
+  }, defaultDocTypes);
+}
+
+async function awaitEachItemInParallel(list, op) {
+  return await Promise.all(list.map(item => op(item)));
+}
+
+export async function resolveIndexPatternConflicts(
+  resolutions,
+  conflictedIndexPatterns,
+  overwriteAll
+) {
+  let importCount = 0;
+  await awaitEachItemInParallel(conflictedIndexPatterns, async ({ obj }) => {
+    let oldIndexId = obj.searchSource.getOwn('index');
+    // Depending on the object, this can either be the raw id or the actual index pattern object
+    if (typeof oldIndexId !== 'string') {
+      oldIndexId = oldIndexId.id;
+    }
+    const newIndexId = resolutions.find(({ oldId }) => oldId === oldIndexId).newId;
+    await obj.hydrateIndexPattern(newIndexId);
+    if (await saveObject(obj, overwriteAll)) {
+      importCount++;
+    }
+  });
+  return importCount;
+}
+
+export async function saveObjects(objs, overwriteAll) {
+  let importCount = 0;
+  await awaitEachItemInParallel(
+    objs,
+    async obj => {
+      if (await saveObject(obj, overwriteAll)) {
+        importCount++;
+      }
+    }
+  );
+  return importCount;
+}
+
+export async function saveObject(obj, overwriteAll) {
+  return await obj.save({ confirmOverwrite: !overwriteAll });
+}
+
+export async function resolveSavedSearches(
+  savedSearches,
+  services,
+  indexPatterns,
+  overwriteAll
+) {
+  let importCount = 0;
+  await awaitEachItemInParallel(savedSearches, async searchDoc => {
+    const obj = await getSavedObject(searchDoc, services);
+    if (!obj) {
+      // Just ignore?
```

Comment:
Seems like something worth logging if we can. We'll probably want to know reasons that this happens.